src/main/java/io/confluent/examples/streams/microservices/EmailService.java

```diff
@@ -40,7 +44,24 @@ public EmailService(final Emailer emailer) {
   public void start(final String bootstrapServers) {
     streams = processStreams(bootstrapServers, "/tmp/kafka-streams");
     streams.cleanUp(); //don't do this in prod as it clears your state stores
+    final CountDownLatch startLatch = new CountDownLatch(1);
+    streams.setStateListener((newState, oldState) -> {
+      if (newState == State.RUNNING && oldState == State.REBALANCING) {
+        startLatch.countDown();
+      }
+
+    });
     streams.start();
+    final boolean rebalanceCompleted;
+
+    try {
+      rebalanceCompleted = startLatch.await(60, TimeUnit.SECONDS);
+      if (!rebalanceCompleted) {
+        throw new RuntimeException("Streams never finished rebalancing on startup");
+      }
+    } catch (final InterruptedException e) {
+       Thread.currentThread().interrupt();
```

Comment:
I first attempted to change the signature of the `Service.start` method to include `throws InterruptedException` and then simply re-throw the `InterruptedException`.   But the changes were going to be more extensive than I felt this fix warranted.  So I'm opting for the next best approach.

This applies to other files as well.